### ISSUE TITLE
genjava should be able to recognize which message files really need rebuilding

### ISSUE THREAD

**Issue by peci1** [rosjava/genjava#3]
At this moment, whenever I have genjava enabled in my workspace, every `catkin_make` call regenerates all the Java msg targets. This takes a lot of unnecessary time (if no message file has changed).

It is needed to implement a behavior similar to the other generators, so that only the message files changed since the last build are rebuilt.

**Comment by stonier** [comment]
Yeah, its a big work though. This is/was a first step along that path and there should be some shortcuts taken (I can't remember exactly what though). The more immediate priority is that the mini gradle projects take too much time and need a better solution.
